src/material_interface.cpp

```diff
@@ -614,6 +615,12 @@ bool material_interface(
                 material_cells.emplace_back(1);
                 material_cells.back()[0] = i;
             }
+            // resolve degenerate cases where all materials are dominated, i.e., one cell bounded by boundaries.
```

Comment:
I think the degenerate case is when none of the material interface intersect with the bounding box of our grid. Only 1 material dominates within the grid bounding box.